CMakeLists.txt

```diff
@@ -881,7 +881,25 @@ if(NOT CMAKE_SYSTEM_NAME STREQUAL "Darwin")
   endif()
 endif()
 
-find_package(PythonInterp REQUIRED)
+# Find Python interpreter. At some point in the future, when Python 3 is
+# preferred over Python 2, all of this can be simply replaced with:
+#
+#     find_package(Python COMPONENTS Interpreter REQUIRED)
+#
+# Until that time, this code looks preferably for version 2 of Python. If not
+# found, version 3 is searched. If none is found a fatal error is raised.
+#
+# This code is essentially the inverse of the built-in CMake FindPython module,
+# which looks preferably for version 3 of Python. If not found, version 2 is
+# searched.
+#
+# https://github.com/Kitware/CMake/blob/cfc92b483fbd3695d4a67843977e709ba4d7ea47/Modules/FindPython/Support.cmake#L2433-L2439
+find_package(Python2 COMPONENTS Interpreter)
+find_package(Python3 COMPONENTS Interpreter)
+
+if (NOT Python2_Interpreter_FOUND AND NOT Python3_Interpreter_FOUND)
+  message(FATAL_ERROR "Could NOT find Python2 or Python3")
+endif()
```

Comment:
If you don't care about the python version here, why not just use `find_package(Python COMPONENTS Interpreter REQUIRED)`?  In fact, I've been trying to explicitly identify the sites that need python2 (as in the gyb handling).